### PR TITLE
Fix stack overflow and support large files

### LINKED ISSUE #12: crash when reading large files
When reading large files, enclave will panic and print ```"thread '<unknown>' has overflowed its stack
fatal runtime error: stack overflow"```.

The crash is caused by ```"fs::read"``` and ```"fs::write"``` apis. 

To fix the issue:
Read/Write data with small buffer.
```rust
loop {
           let result = f.read(&mut buffer);
           ....
} 
```

After the issue is fixed, to support large files:
Change "mesatee_services/fns/sgx_trusted_lib/Enclave.config.xml"  and enlarge ```HeapMaxSize``` and ```StackMaxSize```

### PR DESCRIPTION
**Pull request by litongxin1991**
## Description

Read/Write data with small buffer. 

Fixes #12 

## Type of change (select applied and DELETE the others)

- [X] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
Run PSI examples with large dataset.
Dataset: 100,000,000 data (about 1.5G) for each participant. 
FNS enclave configuration: 
```xml
<StackMaxSize>0x200000</StackMaxSize>
<HeapMaxSize>0x300000000</HeapMaxSize>
```
## Checklist (check ALL before submitting PR, even not applicable)

- [X] Fork the repo and create your branch from `master`.
- [X] If you've added code that should be tested, add tests.
- [X] If you've changed APIs, update the documentation.
- [X] Ensure the tests pass (see CI results).
- [X] Make sure your code lints/format.

### CODE REVIEW

File: mesatee_services/tdfs/internal/client/src/tdfs_client.rs [hunk 1]
```diff
@@ -140,8 +142,16 @@ impl TDFSClient {
         let key_resp = client.request_get_key(&file_info.key_id)?;
         let key_config = key_resp.config;
         let access_path = &file_info.access_path;
-        let ciphertxt = fs::read(access_path)
-            .map_err(|_| mesatee_core::Error::from(mesatee_core::ErrorKind::IoError))?;
+        let mut f = fs::File::open(access_path)?;
+        let mut ciphertxt: Vec<u8> = Vec::new();
```
Comment (m4sterchain) [inline mesatee_services/tdfs/internal/client/src/tdfs_client.rs]: Use with_capacity to initialize the vec.

File: mesatee_services/tdfs/internal/client/src/tdfs_client.rs [hunk 2]
```diff
@@ -140,8 +142,16 @@ impl TDFSClient {
...
+        let mut buffer = vec![0; 1024 * 1024];
```
Comment (m4sterchain) [inline mesatee_services/tdfs/internal/client/src/tdfs_client.rs]: Use with_capacity to initialize the vec.

**Comment by mssun** [comment]
@litongxin1991, could please fix @m4sterchain's comment accordingly? Thanks.

**Comment by litongxin1991** [comment]
> @litongxin1991, could please fix @m4sterchain's comment accordingly? Thanks.

Sure, I will fix it.  Thanks

**Comment by mssun** [comment]
@m4sterchain, please label this PR as "merge when ready" if you think all issues are fixed.

File: mesatee_services/tdfs/internal/client/src/tdfs_client.rs (hunk 1 above)
Comment (litongxin1991) [reply mesatee_services/tdfs/internal/client/src/tdfs_client.rs]: fixed.

**Pull request merged**